.github/workflows/publish-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Publish to PyPI [Test]
+on: [push]
+jobs:
+  test-pypi:
+    name: Create patch version number and push to test-pypi
+    runs-on: ubuntu-latest
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+          installer-parallel: true
+      #----------------------------------------------
+      #       load cached venv if cache exists
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ steps.setup-python.outputs.python-version }}-${{ github.event.repository.name }}-${{ hashFiles('**/poetry.lock') }}
+      #----------------------------------------------
+      # install dependencies if cache does not exist
+      #----------------------------------------------
+      - name: Install dependencies
+        if: steps.cached-poetry-dependencies.outputs.cache-hit != 'true'
+        run: poetry install --no-interaction --no-root
+      #----------------------------------------------
+      # Get the current version and increment it (test-pypi requires a unique version number)
+      #----------------------------------------------
+      - name: Get next version
+        uses: reecetech/version-increment@2022.2.4
```

Comment:
does this auto increment the version after the release if so to which version?

how does it know to increment the minor version or the major version?

Reply:
This step has no side-effects.

Understand this is for test-pypi where versions are ephemeral (periodically deleted). The only requirement for the version we push is that it must be _unique_. This step combines with the following step to generate a guaranteed unique version number for testing purposes.